src/validators/ArrayValidator.ts

```diff
@@ -96,4 +109,12 @@ export class ArrayValidator<T extends unknown[], I = T[number]> extends BaseVali
 			? Result.ok(transformed)
 			: Result.err(new CombinedPropertyError(errors));
 	}
+
+	private isUnique(values: unknown[]) {
+		if (values.length < 2) return true;
+
+		const uniqueArray = uniqWith(values, isDeepStrictEqual);
+
+		return uniqueArray.length === values.length;
+	}
```

Comment:
I'm going to point this out.

Why is this not a constraint? Validators have exactly one task, and should stick to it. Extra steps must run as constraints, as any extra logic in the validators bloat the classes and make them a lot slower.

Also, that code is unworldly inefficient, to start with, it doesn't short-circuit.

Reply:
For the so manieth time, if you want better code then come up with a way for hashing to happen synchronously.